argo.tf

```diff
@@ -25,7 +58,40 @@ locals {
   }
 }
 
-resource "kubernetes_manifest" "this" {
+resource "kubernetes_manifest" "crds" {
+  count = var.enabled && var.argo_enabled && !var.argo_helm_enabled ? 1 : 0
+  manifest = {
+    "apiVersion" = var.argo_apiversion
+    "kind"       = "Application"
+    "metadata" = merge(
+      local.crds_argo_application_metadata,
+      { "name" = "${var.helm_release_name}-crds" },
+      { "namespace" = var.argo_namespace },
+    )
+    "spec" = merge(
+      local.crds_argo_application_values,
+      var.crds_argo_spec
+    )
+  }
+  computed_fields = var.crds_argo_kubernetes_manifest_computed_fields
+
+  field_manager {
+    name            = var.crds_argo_kubernetes_manifest_field_manager_name
+    force_conflicts = var.crds_argo_kubernetes_manifest_field_manager_force_conflicts
+  }
+
+  wait {
+    fields = merge(
+      {
+        "status.sync.status"   = "Synced"
+        "status.health.status" = "Healthy"
```

Comment:
Don't we need to include `status.operationState.phase` as well ?

Reply:
In the past, we had an issue with that status when installing CRDs - I think it was Linkerd. The ArgoCD behavior was different for CRDs.